feat(dags-container): upgrade duckdb to v1.0.0, upgrade base container to use the same as DLM-dags

alee-x opened the pull request:
## :construction: Suggest a change

- Upgrade DuckDB to v1.0.0, no longer rely on bundled static tar file to install duckdb extensions.
- Upgrade container base image to airflow:2.9.0-python3.12 to be in line with DLM-Dags.
- Remove some version pinning from requirements.txt so that only important libraries are version pinned.

## :memo: Pre-merge checklist

Ready to merge? Do not merge until all checks are satisfied.
- [ ] :chart: Have all `required` CI checks passed on the most recent commit?
- [ ] :black_nib: Is the PR title a valid and meaningful conventional-commit message? ie. `type(scope): summary`
- [ ] :boom: Are `breaking changes` declared in the PR title in conventional-commit style? ie. `type!(scope): summary`
- [ ] :art: Does new code follow the code style of this project? 
- [ ] :mag: Has new code been spellchecked and linted?
- [ ] :book: Have docs been updated where necessary?
- [ ] :poop: Have commits been checked for accidental file inclusions?

alee-x committed:
feat(container): upgrade duckdb to v1.0.0, upgrade base container to use the same as DLM-dags

alee-x committed:
fix(repo): stop committing __pycache__ directories

alee-x committed:
feat(container): upgrade ydata profiling to 4.8.3 to fix security alert

alee-x committed:
test(container): split out duckdb install command to find escape char issue

JossWhittle reviewed (dismissed):
Please don't merge this until you've scrubbed out all the pycache directories.

alee-x commented:
> Please don't merge this until you've scrubbed out all the pycache directories.

I thought I'd already done that? The branch this is forked off has all the pycache directories, hence I removed them and updated the gitignore?
